crates/bevy_math/src/primitives/dim3.rs

```diff
@@ -780,6 +780,34 @@ impl Triangle3d {
         ab.cross(ac).length() < 10e-7
     }
 
+    /// Checks if the triangle is acute, meaning all angles are less than 90 degrees
+    #[inline(always)]
+    pub fn is_acute(&self) -> bool {
+        let [a, b, c] = self.vertices;
+        let ab = b - a;
+        let bc = c - b;
+        let ca = a - c;
+
+        // a^2 + b^2 < c^2 for an acute triangle
+        let mut side_lengths = [ab.length(), bc.length(), ca.length()];
```

Comment:
You can avoid unecessary square roots by querying for `length_squared` instead of `length` (since you square the result anyway, and the comparison operations are unaffected by squaring since squaring positive numbers is a rising function)